odoo/tools/misc.py

```diff
@@ -781,19 +781,26 @@ def do_stuff():
     """
     def __init__(self, *loggers):
         self.loggers = loggers
+        self.old_handlers = {}
+        self.old_propagate = {}
+        super().__init__()
 
     def filter(self, record):
         return 0
 
     def __enter__(self):
-        for logger in self.loggers:
-            assert isinstance(logger, str),\
-                "A logger name must be a string, got %s" % type(logger)
-            logging.getLogger(logger).addFilter(self)
+        for logger_name in self.loggers:
+            logger = logging.getLogger(logger_name)
+            self.old_handlers[logger_name] = logger.handlers[:]
+            self.old_propagate[logger_name] = logger.propagate
```

Comment:
Seems unnecessary to have two different maps for what's in essence a logger's configuration. This could be a single entry in a single map. *Copying* the list also seems unnecessary since we're replacing it afterwards (not clearing and updating it).

And since we're messing with the low-level logging information, this should probably be done while holding the logging lock.